keybase/platform_windows.go

```diff
@@ -211,10 +212,149 @@ func (c context) PausedPrompt() bool {
 	return false
 }
 
+// checkRegistryComponents will fall back to read-only if it can't do deletion,
+// returning true if either entries were deleted, or multiple Keybase products were found
+func (c context) checkRegistryComponents(wow64, readonly bool) (result bool) {
+	// e.g.
+	// [HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Installer\UserData\S-1-5-21-2398092721-582601651-115936829-1001\Components\024E69EF1A837C752BFB37F494D86925]
+	// "D6A082CFDEED2984C8688664C76174BC"="C:\\Users\\chris\\AppData\\Local\\Keybase\\Gui\\resources\\app\\images\\icons\\icon-facebook-visibility.gif"
+	// "50DC76D18793BC24DA7D4D681AE74262"="C:\\Users\\chris\\AppData\\Local\\Keybase\\Gui\\resources\\app\\images\\icons\\icon-facebook-visibility.gif"
+
+	var readAccess uint32 = registry.ENUMERATE_SUB_KEYS | registry.QUERY_VALUE
+	if wow64 {
+		readAccess = readAccess | registry.WOW64_32KEY
+	}
+	writeAccess := readAccess | registry.SET_VALUE
+
+	rootName := "SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Installer\\UserData"
+
+	k, err := registry.OpenKey(registry.LOCAL_MACHINE, rootName, readAccess)
+	if err != nil {
+		c.log.Infof("Error opening uninstall subkeys: %s\n", err.Error())
+		return
+	}
+	defer k.Close()
+
+	UIDs, err := k.ReadSubKeyNames(-1)
+	if err != nil {
+		c.log.Infof("Error reading subkeys: %s\n", err.Error())
+		return
+	}
+	for _, UID := range UIDs {
+		componentsKey, err := registry.OpenKey(k, UID+"\\Components", readAccess)
+		if err != nil {
+			c.log.Infof("Error opening subkey %s: %s\n", UID+"\\Components", err.Error())
```

Comment:
This error is just dropped and we charge right through. Is that OK?

Reply:
Don't think so!